src/shared/hooks/useColumnNames.ts

```diff
@@ -0,0 +1,56 @@
+import { Dispatch, SetStateAction } from 'react';
+import { useLocation } from 'react-router-dom';
+import { partition } from 'lodash-es';
+import qs from 'query-string';
+
+import useLocalStorage from './useLocalStorage';
+import useNS from './useNS';
+
+import { Column, nsToDefaultColumns } from '../config/columns';
+import { ColumnConfigurations } from './useColumns';
+
+import { Namespace } from '../types/namespaces';
+import { IDMappingColumn } from '../../tools/id-mapping/config/IdMappingColumnConfiguration';
+import { InvalidParamValue } from '../../uniprotkb/utils/resultsUtils';
+
+const useColumnNames = (
+  namespaceOverride: Namespace | undefined,
+  displayIdMappingColumns?: boolean
+): [
+  Column[],
+  Dispatch<SetStateAction<Column[]>>,
+  boolean,
+  InvalidParamValue | undefined
+] => {
```

Comment:
I think at that point it should probably return an object with sensible field names, because this causes a situation where it's used like
```ts
const [, , , invalidColumns] = useColumnNames(namespaceOverride);
```
in an other file which is a bit strange I think.

If `useViewMode` is similar enough in its approach I'd change it there too.